mono/utils/mono-os-semaphore.h

```diff
@@ -36,13 +36,11 @@
 
 #define MONO_HAS_SEMAPHORES 1
 
-#ifndef NSEC_PER_SEC
-#define NSEC_PER_SEC (1000 * 1000 * 1000)
-#endif
+#define MONO_NSEC_PER_SEC (1000 * 1000 * 1000)
+//usr/include/mach/clock_types.h:86:9: note: previous definition is here
```

Comment:
`NSEC_PER_SEC`  is defined by the  OS.
Mono  defines it itself also.
Hopefully they are the  same thing?
Or  mono should  use  its  own name.

Reply:
Oh  right -- it wasn't  a warning. Just nearby  not great pattern.

Reply:
The comment/warning is if you remove  the ifndef. You  get a   warning because Mono's definition is different  than the OS.
The exact  hazard of the ifndef.

Reply:
`MONO_INFINITE_WAIT`: As long  as they the same, no warning, no error.

Reply:
Yes  I can  remove  the comment too.